libs/ledger-live-common/src/deviceSDK/actions/getDeviceInfo.ts

```diff
@@ -0,0 +1,62 @@
+import { DeviceId, DeviceInfo } from "@ledgerhq/types-live";
+import { Observable } from "rxjs";
+import { scan } from "rxjs/operators";
+import {
+  SharedActionState,
+  getSharedInitialState,
+  sharedReducer,
+} from "./core";
+import {
+  getDeviceInfoTask,
+  GetDeviceInfoTaskError,
+  GetDeviceInfoTaskEvent,
+} from "../tasks/getDeviceInfo";
+
+export type GetDeviceInfoActionArgs = { deviceId: DeviceId };
+
+// Union of all the tasks specific errors
+export type GetDeviceInfoActionErrorType = GetDeviceInfoTaskError;
+
+export type GetDeviceInfoActionState = {
+  deviceInfo: DeviceInfo | null;
+} & SharedActionState<GetDeviceInfoActionErrorType>;
+
+export const initialState: GetDeviceInfoActionState = {
+  deviceInfo: null,
+  ...getSharedInitialState<GetDeviceInfoActionErrorType>(),
+};
+
+export function getDeviceInfoAction({
+  deviceId,
+}: GetDeviceInfoActionArgs): Observable<GetDeviceInfoActionState> {
+  // TODO: to decide: should we push an event if the state is not changing?
```

Comment:
That's a very good question. I'd say no, I think we can only push when the state changes. We can use the distinctUntilChanged operator to do that.